File: Daily Test/Java/Integers - Print Factors 12.9.2025.java
import java.util.*;

public class Main {
    static void fact(int n) {
        List<Integer> big = new ArrayList<>();
        for (int i = 1; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                System.out.print(i + " ");
                if (i != n / i) big.add(n / i);
            }
        }
        Collections.reverse(big);
        for (int v : big) System.out.print(v + " ");
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt(), K = sc.nextInt();
        String[] arr = new String[N];
        for (int i = 0; i < N; i++) arr[i] = sc.next();

        for (int i = 0; i < N; i += K) {
            if (i + K <= N) {
                StringBuilder sb = new StringBuilder();
                for (int j = i; j < i + K; j++)
                    sb.append(arr[j].charAt(arr[j].length() - 1));
                int a = Integer.parseInt(sb.toString());
                if (a == 0) System.out.println("0");
                else { fact(a); System.out.println(); }
            }
        }
    }
}

File: Daily Test/C/Integers - Print Factors 12.9.2025.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

void fact(int n) {
    int tmp[1000], cnt = 0;
    for (int i = 1; i <= sqrt(n); i++) {
        if (n % i == 0) {
            printf("%d ", i);
            if (i != n / i) tmp[cnt++] = n / i;
        }
    }
    for (int i = cnt - 1; i >= 0; i--) printf("%d ", tmp[i]);
}

int main() {
    int N, K;
    scanf("%d %d", &N, &K);
    char arr[N][50];
    for (int i = 0; i < N; i++) scanf("%s", arr[i]);

    for (int i = 0; i < N; i += K) {
        if (i + K <= N) {
            char s[60] = "";
            for (int j = i; j < i + K; j++) {
                int len = strlen(arr[j]);
                char c[2] = {arr[j][len - 1], '\0'};
                strcat(s, c);
            }
            int a = atoi(s);
            if (a == 0) printf("0\n");
            else { fact(a); printf("\n"); }
        }
    }
    return 0;
}